components/webhook-mgt/org.wso2.carbon.identity.webhook.management/src/main/java/org/wso2/carbon/identity/webhook/management/internal/service/impl/WebhookManagementServiceImpl.java

```diff
@@ -186,32 +169,51 @@ public List<Webhook> getWebhooks(String tenantDomain) throws WebhookMgtException
     @Override
     public Webhook activateWebhook(String webhookId, String tenantDomain) throws WebhookMgtException {
 
-        if (LOG.isDebugEnabled()) {
-            LOG.debug(String.format("Activating webhook with ID: %s for tenant: %s",
-                    webhookId, tenantDomain));
-        }
-        int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
-        if (!isWebhookExists(webhookId, tenantId)) {
-            throw WebhookManagementExceptionHandler.handleClientException(
-                    ErrorMessage.ERROR_CODE_WEBHOOK_NOT_FOUND, webhookId);
-        }
-        daoFACADE.activateWebhook(webhookId, tenantId);
-        return daoFACADE.getWebhook(webhookId, tenantId);
+        return processWebhookStatusChange(
+                webhookId, tenantDomain,
+                WebhookStatus.ACTIVE,
+                WebhookStatus.PENDING_ACTIVATION,
+                new WebhookStatus[] {WebhookStatus.INACTIVE, WebhookStatus.PENDING_DEACTIVATION},
+                (id, tid, wh) -> daoFACADE.activateWebhook(id, tid, wh.getEventsSubscribed(), wh.getStatus(),
+                        wh.getEndpoint()),
+                ErrorMessage.ERROR_CODE_WEBHOOK_ACTIVATION_NOT_ALLOWED_ERROR
+        );
     }
 
     @Override
     public Webhook deactivateWebhook(String webhookId, String tenantDomain) throws WebhookMgtException {
 
-        if (LOG.isDebugEnabled()) {
-            LOG.debug(String.format("Deactivating webhook with ID: %s for tenant: %s",
-                    webhookId, tenantDomain));
-        }
+        return processWebhookStatusChange(
+                webhookId, tenantDomain,
+                WebhookStatus.INACTIVE,
+                WebhookStatus.PENDING_DEACTIVATION,
+                new WebhookStatus[] {WebhookStatus.ACTIVE, WebhookStatus.PENDING_ACTIVATION},
+                (id, tid, wh) -> daoFACADE.deactivateWebhook(id, tid, wh.getEventsSubscribed(), wh.getStatus(),
+                        wh.getEndpoint()),
+                ErrorMessage.ERROR_CODE_WEBHOOK_DEACTIVATION_NOT_ALLOWED_ERROR
+        );
+    }
```

Comment:
my comments above applies here as well.

And also better keep clear contracts in facade/data layer for activate deactivate operations

Reply:
Refactored and improved the readability
fixed with https://github.com/wso2/carbon-identity-framework/pull/6885/commits/fb575c056228c93d2221b4ea29c500a76d09311c